gwt-datetimeformat/src/test/java/org/gwtproject/i18n/jvm/DateTimeFormat_de_Test.java

```diff
@@ -0,0 +1,207 @@
+    date.setSeconds(10);
+    assertEquals("Q1 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 1, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q1 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 2, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q1 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 3, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q2 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 4, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q2 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 5, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q2 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 6, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q3 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 7, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q3 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 8, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q3 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 9, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q4 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 10, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q4 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+    date = new Date(2006 - 1900, 11, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("Q4 2006", DateTimeFormat.getFormat("QQ yyyy").format(date));
+  }
+
+  public void test_quote() {
+    Date date = new Date(2006 - 1900, 6, 27);
+    date.setHours(13);
+    date.setMinutes(10);
+    date.setSeconds(10);
+    assertEquals("13 o'clock", DateTimeFormat.getFormat("HH 'o''clock'").format(date));
+    assertEquals("13 oclock", DateTimeFormat.getFormat("HH 'oclock'").format(date));
+    assertEquals("13 '", DateTimeFormat.getFormat("HH ''").format(date));
+  }
+
+  public void test_yyyyyMMMMM() {
+    Date date = new Date(2006 - 1900, 6, 27, 13, 10, 10);
+    assertEquals(
+        "02006.J.27 n. Chr. 01:10 PM",
+        DateTimeFormat.getFormat("yyyyy.MMMMM.dd GGG hh:mm aaa").format(date));
+  }
+
+  //  public void testMessageDateTime() {
```

Comment:
?

Reply:
those tests requires the migration of the MessageConstants so  i kept them till we have that part ready